src/app/services/auth.service.ts

```diff
@@ -1,41 +1,58 @@
-import { Injectable, inject, signal } from '@angular/core';
+import { Injectable, inject } from '@angular/core';
 import {
   Auth,
+  UserCredential,
   createUserWithEmailAndPassword,
   signInWithEmailAndPassword,
   signOut,
   updateProfile,
   user,
 } from '@angular/fire/auth';
-import { Observable, Subject, from } from 'rxjs';
-import { AuthUser } from '../interfaces/authInterface';
+import { Router } from '@angular/router';
+import { Observable, from, map } from 'rxjs';
+import { UsersFirebaseService } from './users-firebase.service';
 
 @Injectable({
   providedIn: 'root',
 })
 export class AuthService {
   firebaseAuth = inject(Auth);
   user$ = user(this.firebaseAuth);
-  currentUserSig = signal<AuthUser | null | undefined>(undefined);
-  isUserSet$ = new Subject<boolean>();
+  usersFirebaseService = inject(UsersFirebaseService);
+  router = inject(Router);
 
   register(
     email: string,
     username: string,
     password: string
-  ): Observable<void> {
+  ): Observable<{ username: string; email: string }> {
+    this.usersFirebaseService.addUser(username, email);
     return from(
       createUserWithEmailAndPassword(this.firebaseAuth, email, password).then(
-        (response) => updateProfile(response.user, { displayName: username })
+        (response) => {
+          const username = response?.user?.displayName || '';
+          const email = response?.user?.email || '';
+          updateProfile(response.user, { displayName: username });
+          this.router.navigate(['/home']);
```

Comment:
In the flow of the state management, I am not sure if the services should be 'speaking' to the UI via the router. I think the user should have an observer for this service so that the UI is not directly changed by a service